src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.cs

```diff
@@ -97,5 +97,24 @@ private void VerifyMutable()
                 ThrowHelper.ThrowInvalidOperationException_TypeInfoResolverImmutable();
             }
         }
+
+        internal static DefaultJsonTypeInfoResolver DefaultInstance
+        {
+            [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
+            [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
+            get
+            {
+                if (s_defaultInstance is DefaultJsonTypeInfoResolver resolver)
```

Comment:
```suggestion
                if (s_defaultInstance != null)
```

Reply:
It's intentional binding `resolver` so it can be returned without the need to re-read the static.